### PR TITLE
Add support for crossgen2 to jit-dasm and jit-diff.

### PR DESCRIPTION
**Pull request by erozenfeld**
--crossgen arg can now point to crossgen.exe or crossgen2.exe.

crossgen.exe and crossgen2.exe have diffferent command line args
so jit-dasm takes care of passing the right args depending on whether
--crossgen points to crossgen.exe or crossgen2.exe.

### CODE REVIEW

**Comment by erozenfeld** [comment]
@dotnet/jit-contrib PTAL

File: src/jit-dasm/jit-dasm.cs [hunk 1]
```diff
@@ -109,6 +116,20 @@ private void Validate()
...
+                    if (fullCrossgenPathLower.EndsWith("crossgen.exe"))
+                    {
+                        _codeGenerator = CodeGenerator.Crossgen;
+                    }
+                    else if (fullCrossgenPathLower.EndsWith("crossgen2.exe"))
+                    {
+                        _codeGenerator = CodeGenerator.Crossgen2;
+                    }
+                    else
+                    {
```
Comment (echesakovMSFT) [inline src/jit-dasm/jit-dasm.cs]: This wouldn't work on Linux/macOS

File: src/jit-dasm/jit-dasm.cs [hunk 2]
```diff
@@ -560,15 +588,145 @@ void AddEnvironmentVariable(string varName, string varValue)
...
+            abstract protected void AddAssembliesPathsOption(List<string> commandArgs);
+
+            abstract protected void AddOutputPathOption(List<string> commandArgs, string outputPath);
+
+            abstract protected void AddOptimizationOption(List<string> commandArgs);
+
+            abstract protected Command CreateCommand(ICommandResolverPolicy resolverPolicy, List<string> commandArgs);
+        }
+
+        private class CrossgenDisasmEngine : DisasmEngine
```
Comment (echesakovMSFT) [inline src/jit-dasm/jit-dasm.cs]: Should the classes should be `sealed`?

File: src/jit-dasm/jit-dasm.cs (hunk 1 above)
Comment (erozenfeld) [reply src/jit-dasm/jit-dasm.cs]: Good catch, thanks. Fixed.

File: src/jit-dasm/jit-dasm.cs (hunk 2 above)
Comment (erozenfeld) [reply src/jit-dasm/jit-dasm.cs]: Added sealed.

**Comment by erozenfeld** [comment]
I also changed command line list creation to use Add instead of Insert to make it linear instead of quadratic.

**Pull request merged**